robot_skills/src/robot_skills/robot.py

```diff
@@ -68,6 +68,9 @@ def __init__(self, robot_name="", wait_services=False):
         # Reasoning/world modeling
         self.parts['ed'] = world_model_ed.ED(self.robot_name, self.tf_listener)
 
+        # Ignore diagnostics: parts that are not present in the real robot
+        self._ignored_parts = []
```

Comment:
As a general rule of thumb: If you search in a container, don't use a list, since on average you have to inspect half the elements in it, and for a miss you have to check every element. Instead, use a dict or a set. (like `self._ignored_parts = set([...])`)